src/container/internal.tsx

```diff
@@ -74,7 +78,10 @@ export default function InternalContainer({
   const hasDynamicHeight = isRefresh && variant === 'full-page';
   const overlapElement = useDynamicOverlap({ disabled: !hasDynamicHeight || !__darkHeader });
 
-  const mergedRef = useMergeRefs(rootRef, subStepRef, __internalRootRef);
+  const [breakpoint, breakpointRef] = useContainerBreakpoints(BREAKPOINTS_DESCENDING.filter(bp => bp !== 'default'));
```

Comment:
Would it make sense to move the `useContainerBreakpoints` into `useMedia`?

Reply:
also, instead of listening to all breakpoints (except default), could you extract the list of breakpoints use from the `media` prop and use them instead?